.github/workflows/release.yml

```
@@ -0,0 +1,42 @@
+name: Semantic Release
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read # for checkout
+
+jobs:
+  release:
+    name: Semantic Release
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write # to be able to publish a GitHub release
+      issues: write # to be able to comment on released issues
+      pull-requests: write # to be able to comment on released pull requests
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        uses: pnpm/action-setup@v4
```

cf. https://github.com/pnpm/action-setup?tab=readme-ov-file#use-cache-to-reduce-installation-time

```suggestion
        uses: pnpm/action-setup@v4
        with:
          run-install: false
```